oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -311,6 +342,13 @@ public UserCredentials getCredentialsFromCode(
         requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), tokenContent);
     tokenRequest.setParser(new JsonObjectParser(OAuth2Utils.JSON_FACTORY));
 
+    if (this.clientAuthenticationType == ClientAuthenticationType.CLIENT_SECRET_BASIC) {
+      String encodedCredentials =
+          BaseEncoding.base64()
+              .encode((clientId.getClientId() + ":" + clientId.getClientSecret()).getBytes());
```

Comment:
Can you pull this into a helper function that can be unit tested?

Reply:
Ideally it takes the client id and secret as params, so a unit test can hard code expect values.